src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py

```diff
@@ -487,3 +492,100 @@ def _stream_trace(streaming_url, user_name, password):
                   .encode(std_encoding, errors='replace')
                   .decode(std_encoding, errors='replace'), end='') # each line of log has CRLF.
     r.release_conn()
+
+def upload_ssl_cert(resource_group_name, name, certificate_password, certificate_file):
+    client = web_client_factory()
+    webapp = client.sites.get_site(resource_group_name, name)
+    cert_file = open(certificate_file, 'rb')
+    cert_contents = cert_file.read()
+    hosting_environment_profile_param = webapp.hosting_environment_profile
+    if hosting_environment_profile_param is None:
+        hosting_environment_profile_param = ""
+
+    thumb_print = get_cert(certificate_password, certificate_file)
+    cert_name = generate_cert_name(thumb_print, hosting_environment_profile_param,
+                                   webapp.location, resource_group_name)
+    cert_base64_str = base64.b64encode(cert_contents).decode("utf-8")
+    cert = Certificate(password=certificate_password, pfx_blob=cert_base64_str,
+                       location=webapp.location)
+    return client.certificates.create_or_update_certificate(resource_group_name, cert_name, cert)
+
+def generate_cert_name(thumb_print, hosting_environment, location, resource_group_name):
+    return "%s_%s_%s_%s" % (thumb_print, hosting_environment, location, resource_group_name)
+
+def get_cert(certificate_password, certificate_file):
+    ''' Decrypts the .pfx file '''
+    p12 = OpenSSL.crypto.load_pkcs12(open(certificate_file, 'rb').read(), certificate_password)
+    cert = p12.get_certificate()
+    digest_algorithm = 'sha1'
+    thumbprint = cert.digest(digest_algorithm).decode("utf-8").replace(':', '')
+    return thumbprint
+
+def list_ssl_certs(resource_group_name):
+    client = web_client_factory()
+    return client.certificates.get_certificates(resource_group_name)
+
+def delete_ssl_cert(resource_group_name, certificate_thumbprint):
+    client = web_client_factory()
+    cert_name = None
+    webapp_certs = client.certificates.get_certificates(resource_group_name)
+    for webapp_cert in webapp_certs:
+        if webapp_cert.thumbprint == certificate_thumbprint:
+            cert_name = webapp_cert.name
+
+    return client.certificates.delete_certificate(resource_group_name, cert_name)
+
+def should_use_deployment_slot(webapp_name, slot=None):
+    qualified_site_name = None
+    if slot is not None:
+        qualified_site_name = "%s(%s)" % (webapp_name, slot)
```

Comment:
Instead of using `%`, use string.format()
see http://stackoverflow.com/a/12382738 & https://docs.python.org/release/3.1.5/library/stdtypes.html#old-string-formatting-operations